run/jobs/main.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import random
+import sys
+import time
+
+# Retrieve Job-defined env vars
+TASK_NUM = os.getenv('TASK_NUM') or 0
+ATTEMPT_NUM = os.getenv('ATTEMPT_NUM') or 0
+# Retrieve User-defined env vars
+SLEEP_MS = os.getenv('SLEEP_MS')
+FAIL_RATE = os.getenv('FAIL_RATE')
+
+
+# Define main script
+def main(sleep_ms=None, fail_rate=None):
```

Comment:
I've set it up this way to make testing easier of the main function, so instead of setting env vars for the test I can just pass them in as `main.main(SLEEP_MS, RATE)`. Therefore I have to check for Nones anyways. Is there something that is more idiomatic?

Reply:
I was thinking that if fail_rate defaults to 0, then the logic in the random_failure function could be simplified.